projects/noloan/app/filter/SpamFilterAlgorithm.java

```diff
@@ -0,0 +1,5 @@
+package projects.noloan.app.filter;
```

Comment:
I don't think we'll need 2 algorithms at the same time, we'll just evolve the one we have.
So no need for an interface.